packages/lib/src/vis/shared/PanZoomMesh.tsx

```diff
@@ -1,23 +1,27 @@
 import { useThree } from '@react-three/fiber';
-import type { ThreeEvent } from '@react-three/fiber/dist/declarations/src/core/events';
+import type { ThreeEvent } from '@react-three/fiber';
 import { clamp } from 'lodash';
 import { useRef, useCallback, useEffect } from 'react';
 import { Vector2, Vector3 } from 'three';
 
 import { useWheelCapture } from '../hooks';
 import type { ModifierKey } from '../models';
-import { CAMERA_TOP_RIGHT, isEventValid } from '../utils';
+import { CAMERA_TOP_RIGHT, isEventValid, noModifierKeyPressed } from '../utils';
 import { useAxisSystemContext } from './AxisSystemContext';
 
 const ZOOM_FACTOR = 0.95;
 const ONE_VECTOR = new Vector3(1, 1, 1);
 
 interface Props {
+  pan?: boolean;
   panKey?: ModifierKey;
+  zoom?: boolean;
+  zoomKeyX?: ModifierKey;
+  zoomKeyY?: ModifierKey;
 }
 
 function PanZoomMesh(props: Props) {
-  const { panKey } = props;
+  const { pan = true, zoom = true, panKey, zoomKeyX, zoomKeyY } = props;
```

Comment:
Seeing this API in practice now, I realize that it's a bit unclear that `zoomKeyX` and `zoomKeyY` serve to enable an additional feature on top of `zoom` - i.e. it's not because `zoom` is `true` that the zoom on X/Y is enabled. This is a bit inconsistent with the behaviour of the `panKey` prop, which does not enable a new feature - it just configures a keybinding. In other words, I feel like configuring the keybindings for zooming on X/Y should be possible independently of enabling zooming on X/Y.

Perhaps we should provide two additional boolean props for this. Doing so would allow us to set sane defaults for `zoomKeyX` and `zoomKeyY` so consumers could just write `<PanZoomMesh zoomX zoomY />` without worrying about key bindings. It would also simplify your `isEventValid` logic.

When it comes to naming, since `zoomXKey` is a bit ugly, I propose `xZoom` and `xZoomKey`.

What do you think?

Reply:
If we go this way, I would like to clarify a few usecases. Tell me if you think this is right or not:
- If setting only `zoomX` to `true` _without a key_, the wheel (with no key pressed) should zoom only in X and not Y
- If setting only `zoomX` to `true` _with a key_, the wheel (with no key pressed) should  **not** allow to zoom. It should zoom in X **only** if the wheel is scrolled with the key pressed.
- If setting `zoomX` and `zoomY` to `true` with keys for both, we agreed that the wheel (with no key pressed) should zoom in X and Y.
- If setting `zoomX` to `true` _with a key_ and `zoomY` to `true` _without a key_, the wheel (with no key pressed) should still zoom in X and Y. The only axial zoom allowed is the X zoom when pressing the corresponding key when scrolling the wheel.

If we agree on the behaviour above, I don't see how it will simplify the logic. Or did I miss something ?

Reply:
I was suggesting that the x/y zoom keys would always be set (because we'd give them the default values `Alt` and `Shift`). The goal was solely to make the API clearer by making the activation of the x-only and y-only zoom features clearer. I was _not_ suggesting that enabling either x-only or y-only zoom without a key would disable the global x/y zoom. Sorry for the confusion.

Enabling axial zooming in one direction without a modifier key is a use case (perhaps interesting for line visualizations) that we hadn't considered before, I think. It definitely complexifies the problem. 😅 Perhaps it's not worth worrying about for now.

Reply:
Here are all the use cases I can think of:

| Use case | Current API | Proposed API |
| --- | --- | --- |
| No global zoom, no axial zoom | `zoom: false` | `zoom: false` |
| Global zoom only (no modifier key) | `zoom: true` | `zoom: true` |
| Global zoom + x-only zoom with `Alt` +  | `zoom: true, zoomKeyX: 'Alt'` | `zoom: true, xZoom: true` |
| Global zoom + y-only zoom with `Shift` +  | `zoom: true, zoomKeyY: 'Shift'` | `zoom: true, yZoom: true` |
| Global zoom + x-only zoom with `Alt` + y-only zoom with `Shift` +  | `zoom: true, zoomKeyX: 'Alt', zoomKeyY: 'Shift'` | `zoom: true, xZoom: true, yZoom: true` |
| Global zoom + x-only zoom with `Shift` + y-only zoom with `Alt` +  | `zoom: true, zoomKeyX: 'Shift', zoomKeyY: 'Alt'` | `zoom: true, xZoom: true, xZoomKey: 'Shift', yZoom: true, yZoomKey: 'Alt'` |
| Global zoom with `Ctrl` | not supported | not supported |
| X-only zoom (no modifier key, no global zoom) | not supported | not supported |
| Y-only zoom (no modifier key, no global zoom) | not supported | not supported |
| X-only zoom (no modifier key, no global zoom) + Y-only zoom with `Ctrl` | not supported | not supported |
| Y-only zoom (no modifier key, no global zoom) + X-only zoom with `Ctrl` | not supported | not supported |

Reply:
Thanks, that's way clearer. I think the implementation should now match this